src/db/mod.rs

```diff
@@ -129,45 +129,26 @@ impl Database {
         self.inner.write_concern.as_ref()
     }
 
-    /// Gets a handle to a collection specified by `name` of the database. The `Collection` options
-    /// (e.g. read preference and write concern) will default to those of the `Database`.
-    ///
-    /// This method does not send or receive anything across the wire to the database, so it can be
-    /// used repeatedly without incurring any costs from I/O.
-    pub fn collection(&self, name: &str) -> Collection {
-        Collection::new(self.clone(), name, None)
-    }
-
     /// Gets a handle to a collection with type `T` specified by `name` of the database. The
     /// `Collection` options (e.g. read preference and write concern) will default to those of the
     /// `Database`.
     ///
     /// This method does not send or receive anything across the wire to the database, so it can be
     /// used repeatedly without incurring any costs from I/O.
-    pub fn collection_with_type<T>(&self, name: &str) -> Collection<T>
+    pub fn collection<T>(&self, name: &str) -> Collection<T>
```

Comment:
I know we discussed the possibility of introducing a `collection_with_document` helper (or something like that) as an ergonomic way to retrieve a `Collection<Document>`, but after writing a few of these conversions, it seemed simpler / easier to just use the turbofish (`::<>`) operator on the existing helpers.

e.g.
```rust
let collection: Collection<Document> = db.collection("foo"); // explicit type way
let collection = db.collection::<Document>("foo"); // turbofish
let collection = db.collection_with_document("foo"); // helper
```

To me, the helper didn't really make it any easier to type nor did it actually make it clearer what was happening, so it seemed fine to me to just not have it. Thoughts?

Reply:
from my outsider perspective it doesn't seem to make a significant difference, but I'd defer to you two on whether `collection_with_document` is in any way more idiomatic